core/base/marchingTetrahedra/MarchingTetrahedra.h

```diff
@@ -0,0 +1,2155 @@
+     * @param[in] triangulation Triangulation
+     * @return int
+     */
+    template <typename triangulationType>
+    int writeSeparators_3D(const unsigned char *const tetCases,
+                           const size_t *numTriangles,
+                           const unsigned long long *const scalars,
+                           const triangulationType &triangulation);
+
+    /**
+     * Writes the geometry of all Triangles to the ouputVariables set by
+     * setOutput. 3D basin boundaries.
+     *
+     * @tparam triangulationType Triangulationtype
+     * @param[in] tetCases Binary codes
+     * @param[in] numTriangles Number of triangles generated per thread
+     * @param[in] scalars Scalars
+     * @param[in] triangulation Triangulation
+     * @return int
+     */
+    template <typename triangulationType>
+    int writeBoundaries_3D(const unsigned char *const tetCases,
+                           const size_t *numTriangles,
+                           const unsigned long long *const scalars,
+                           const triangulationType &triangulation);
+
+    /**
+     * Writes the geometry of all Triangles to the ouputVariables set by
+     * setOutput. 3D detailed basin boundaries.
+     *
+     * @tparam triangulationType Triangulationtype
+     * @param[in] tetCases Binary codes
+     * @param[in] numTriangles Number of triangles generated per thread
+     * @param[in] scalars Scalars
+     * @param[in] triangulation Triangulation
+     * @return int
+     */
+    template <typename triangulationType>
+    int writeBoundariesDetailed_3D(const unsigned char *const tetCases,
+                                   const size_t *numTriangles,
+                                   const unsigned long long *const scalars,
+                                   const triangulationType &triangulation);
+
+  protected:
+    // Output options
+    SURFACE_MODE SurfaceMode{SURFACE_MODE::SM_SEPARATORS};
+
+    // Output data
+    SimplexId output_numberOfPoints_{};
+    SimplexId output_numberOfCells_{};
+    std::vector<float> output_points_;
+    std::vector<unsigned long long> output_cells_labels_;
+    std::vector<SimplexId> output_cells_connectivity_;
+  };
+} // namespace ttk
+
+template <typename dataType, typename triangulationType>
+int ttk::MarchingTetrahedra::execute(const dataType *const scalars,
+                                     const triangulationType &triangulation) {
+
+  Timer t;
+
+  if(scalars == nullptr)
+    return this->printErr("Input scalar field pointer is null.");
+
+  const SimplexId nV = triangulation.getNumberOfVertices();
+  const SimplexId nC = triangulation.getNumberOfCells();
+  const int dim = triangulation.getDimensionality();
+
+  std::vector<unsigned char> tetCases;
+  std::vector<size_t> numberOfTetCases;
+  std::vector<unsigned long long> cScalars;
+
+  tetCases.resize(nC);
+  cScalars.resize(nV);
+
+#ifdef TTK_ENABLE_OPENMP
+  numberOfTetCases.resize(omp_get_max_threads());
```

Comment:
I believe you want to use `threadNumber_` instead of `omp_get_max_threads()`

Reply:
You are totally right. The code now also consistently uses `this->threadNumber_`.

Reply:
wonderful.
it seems that your latest commits don't build under the CI.
could you have a look into that please?
thanks!